multiview/embed/omnibus.py

```diff
@@ -0,0 +1,133 @@
+
+        Parameters
+        ----------
+        n_components : strictly positive int (default = 2)
+            Desired dimensionality of output embeddings. See graspy docs for
+            additional details.
+
+        distance_metric : string (default = 'euclidean')
+            Distance metric used to compute pairwise distances.
+
+        normalize : string or None (default = 'l1')
+            Normalize function to use on views before computing
+            pairwise distances. Must be 'l2', 'l1', 'max'
+            or None. If None, the distance matrices will not be normalized.
+
+        algorithm : string (default = 'randomized')
+            SVD solver to use. Must be 'full', 'randomized', or 'truncated'.
+            See graspy docs for details.
+
+        n_iter : positive int (default = 5)
+            Number of iterations for randomized SVD solver. See graspy docs for
+            details.
+        """
+        super().__init__()
+        self.n_components = n_components
+        self.normalize = normalize
+        self.distance_metric = distance_metric
+        self.algorithm = algorithm
+        self.n_iter = n_iter
+        self.check_params()
+
+    def check_params(self):
+        """
+        Checks that Omnibus arguments are valid. A ValueError
+        is thrown if any are not. The checks performed are:
+            - distance_metric is valid
+            - algorithm is valid
+            - normalize is valid
+            - n_components is positive int
+            - n_iter is positive int
+        """
+
+        valid_metrics = ['braycurtis', 'canberra',
+                         'chebyshev', 'cityblock', 'correlation',
+                         'cosine', 'dice', 'euclidean', 'hamming',
+                         'jaccard', 'jensenshannon', 'kulsinski',
+                         'mahalanobis', 'matching', 'minkowski',
+                         'rogerstanimoto', 'russellrao', 'seuclidean',
+                         'sokalmichener', 'sokalsneath', 'sqeuclidean', 'yule']
+
+        valid_algorithms = ["randomized", "full", "truncated"]
+
+        valid_normalize = ["l1", "l2", "max"]
+
+        if self.distance_metric not in valid_metrics:
+            raise ValueError("distance_metric must be in \
+                             scipy.spatial.distances.pdist.")
+
+        if self.algorithm not in valid_algorithms:
+            raise ValueError("algorithm must be 'randomized', \
+                            'full', or 'truncated'.")
+
+        if self.normalize is not None and \
+           self.normalize not in valid_normalize:
+            raise ValueError("normalize must be 'l2', 'l1', or 'max'.")
+
+        if not isinstance(self.n_components, int) or self.n_components <= 0:
+            raise ValueError("n_components must be positive int.")
+
+        if not isinstance(self.n_iter, int) or self.n_iter <= 0:
+            raise ValueError("n_iter must be positive int.")
+
+    def fit_transform(self, Xs):
```

Comment:
this class should also have `fit`, and should save embeddings as attributes

Reply:
Fixed